provider-ui/components/ProviderComponent.js

```diff
@@ -118,89 +141,139 @@ class ProviderComponent extends React.Component {
     };
   }
 
+  handleModalOpen() {
+    const self = this;
+    return () => {
+      self.setState({ modalOpen : true });
+    };
+  }
+
+  handleModalClose() {
+    const self=this;
+    return () => {
+      self.setState({ modalOpen : false });
```

Comment:
i guess we don't need to use 
```js
const self = this;
```
Since this is an arrow function lexical scoping of `this` happens here

so you could write 

```js
handleModalClose() {
       return () => {
      this.setState({ modalOpen : false });
   }
```
and it would still work